Src/FluentAssertions/EventRaisingExtensions.cs

```diff
@@ -53,49 +53,43 @@ public static IEventRecording WithSender(this IEventRecording eventRecording, ob
         }
 
         /// <summary>
-        /// Asserts that at least one occurrence of the events had at least one of the arguments matching a predicate. Returns
-        /// only the events that matched that predicate.
+        /// Asserts that at least one of the occurred events is of the expected type <typeparamref name="T"/>
+        /// and optionally has a matching predicate. Returns
+        /// only the events that matched those type and predicate.
         /// </summary>
         public static IEventRecording WithArgs<T>(this IEventRecording eventRecording, Expression<Func<T, bool>> predicate)
         {
             Guard.ThrowIfArgumentIsNull(predicate, nameof(predicate));
 
             Func<T, bool> compiledPredicate = predicate.Compile();
 
-            bool hasArgumentOfRightType = false;
             var eventsMatchingPredicate = new List<OccurredEvent>();
 
             foreach (OccurredEvent @event in eventRecording)
             {
                 var typedParameters = @event.Parameters.OfType<T>().ToArray();
-                if (typedParameters.Any())
-                {
-                    hasArgumentOfRightType = true;
-                }
 
                 if (typedParameters.Any(parameter => compiledPredicate(parameter)))
                 {
                     eventsMatchingPredicate.Add(@event);
                 }
             }
 
-            if (!hasArgumentOfRightType)
-            {
-                throw new ArgumentException("No argument of event " + eventRecording.EventName + " is of type <" + typeof(T) + ">.");
-            }
+            bool hasFoundArgumentsBefore = eventsMatchingPredicate.Any();
```

Comment:
How about naming this `foundMatchingEvent` or `foundEventMatchingPredicate`?

Reply:
Yeah.. can agree with that

Reply:
I'm sorry, I only meant renaming `hasFoundArgumentsBefore` to either `foundMatchingEvent` _or_ `foundEventMatchingPredicate`.

`eventsMatchingPredicate` shouldn't be renamed.